Source/WebCore/platform/graphics/cocoa/IOSurface.mm

```diff
@@ -685,6 +689,28 @@ static IntSize computeMaximumSurfaceSize()
     m_colorSpace = surfaceColorSpace().value_or(DestinationColorSpace::SRGB());
 }
 
+#if HAVE(SUPPORT_HDR_DISPLAY)
+void IOSurface::setContentEDRHeadroom(float headroom)
+{
+    if (m_contentEDRHeadroom && headroom == m_contentEDRHeadroom)
+        return;
+
+    LOG_WITH_STREAM(HDR, stream << "IOSurface::setContentEDRHeadroom " << this << " " << headroom);
+    m_contentEDRHeadroom = headroom;
+    IOSurfaceSetValue(m_surface.get(), CFSTR("IOSurfaceContentHeadroom"), @(headroom));
+}
+
+float IOSurface::contentEDRHeadroom() const
+{
+    if (!m_contentEDRHeadroom) {
+        m_contentEDRHeadroom = 1;
+        if (auto valueNumber = dynamic_cf_cast<CFNumberRef>(IOSurfaceCopyValue(m_surface.get(), CFSTR("IOSurfaceContentHeadroom"))))
```

Comment:
I think you need an `adoptCF` in here since presumably `IOSurfaceCopyValue` retains.